ansible/plugins/action/fetch_no_slurp.py

```diff
@@ -0,0 +1,160 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import os
+from ansible.errors import AnsibleError, AnsibleActionFail, AnsibleActionSkip
+from ansible.module_utils.common.text.converters import to_bytes, to_text
+from ansible.module_utils.six import string_types
+from ansible.module_utils.parsing.convert_bool import boolean
+from ansible.plugins.action import ActionBase
+from ansible.utils.display import Display
+from ansible.utils.hashing import checksum, md5, secure_hash
+from ansible.utils.path import makedirs_safe, is_subpath
+
+display = Display()
+
+
+class ActionModule(ActionBase):
+
+    def run(self, tmp=None, task_vars=None):
```

Comment:
Why do we need a tmp here? Is it an official signature designed by Ansible?

Reply:
Yes this is official signature from their method interface. I believe this is for backward compatibility usage: https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/action/__init__.py#L91

FYI: most of the code here is identical to https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/action/fetch.py with some modification to remove their use of slurp and force use `fetch_file`